src/containers/course-section/resource-item/ResourceItem.tsx

```diff
@@ -111,14 +107,18 @@ const ResourceItem: FC<ResourceItemProps> = ({
 
   const setAvailabilityStatus = useCallback(
     (status: ResourceAvailabilityStatusEnum) => {
-      updateAvailability?.(resource, {
+      return updateAvailability?.(resource, {
         status,
         date: null
       })
     },
     [resource, updateAvailability]
   )
-
+  useEffect(() => {
+    if (resourcesAvailability === ResourcesAvailabilityEnum.OpenManually) {
+      setAvailabilityStatus(ResourceAvailabilityStatusEnum.Closed)
+    }
+  }, [resourcesAvailability, setAvailabilityStatus])
```

Comment:
Never use `useEffect` with updating state inside. It will cause an infinite loop

https://github.com/user-attachments/assets/f70cdefc-5c48-4949-9fa6-184a9e3d44d3

Reply:
this useEffect was used before mine. I only fixed eslint using. But thanks, I`ll check it